File: src/tests/DataBlockASPNETDemo/vb.net/mappings/Book.vb
Imports System
Imports System.Data
Imports voidsoft.DataBlock
Namespace BookManager.Mappings

	<Serializable()> _
	Public Class Book
	Inherits TableMetadata
		Private _fields As DatabaseField()

		Public Sub New()
            _fields = New DatabaseField(5) {}
			_fields(0) = New DatabaseField(DbType.Int32, "BookId", True, True, Nothing)
			_fields(1) = New DatabaseField(DbType.String, "Name", False, False, Nothing)
			_fields(2) = New DatabaseField(DbType.Int32, "Pages", False, False, Nothing)
			_fields(3) = New DatabaseField(DbType.String, "ISBN", False, False, Nothing)
			_fields(4) = New DatabaseField(DbType.String, "Genre", False, False, Nothing)
			_fields(5) = New DatabaseField(DbType.Int16, "Grade", False, False, Nothing)
			Me.currentTableName = "Book"
			Me.listRelations.Add(New ParentTableRelation("BookAuthors", "BookId", TableRelationCardinality.OneToMany, True))
			Me.listRelations.Add(New ManyToManyTableRelation("Author", "BookAuthors", "BookId", "AuthorId"))
		End Sub

		Public Overloads Overrides Property TableFields() As DatabaseField()
			Get
				Return _fields
			End Get
			Set
				_fields = value
			End Set
		End Property

		Public Property BookId() As System.Int32
			Get
				Return CType((Me.GetField("BookId")).fieldValue, System.Int32)
			End Get
			Set
				Me.SetFieldValue("BookId", value)
			End Set
		End Property

		Public Property Name() As System.String
			Get
				Return CType((Me.GetField("Name")).fieldValue, System.String)
			End Get
			Set
				Me.SetFieldValue("Name", value)
			End Set
		End Property

		Public Property Pages() As System.Int32
			Get
				Return CType((Me.GetField("Pages")).fieldValue, System.Int32)
			End Get
			Set
				Me.SetFieldValue("Pages", value)
			End Set
		End Property

		Public Property ISBN() As System.String
			Get
				Return CType((Me.GetField("ISBN")).fieldValue, System.String)
			End Get
			Set
				Me.SetFieldValue("ISBN", value)
			End Set
		End Property

		Public Property Genre() As System.String
			Get
				Return CType((Me.GetField("Genre")).fieldValue, System.String)
			End Get
			Set
				Me.SetFieldValue("Genre", value)
			End Set
		End Property

		Public Property Grade() As System.Int16
			Get
				Return CType((Me.GetField("Grade")).fieldValue, System.Int16)
			End Get
			Set
				Me.SetFieldValue("Grade", value)
			End Set
		End Property

		Public Function GetBookAuthors() As Array
			Dim relatedType As Type = Type.GetType("BookManager.Mappings.BookAuthors")
			Return Me.GetRelatedTableData(relatedType)
		End Function

		Public Function GetAuthor() As Array
			Dim relatedType As Type = Type.GetType("BookManager.Mappings.Author")
			Dim intermediaryType As Type = Type.GetType("BookManager.Mappings.BookAuthors")
			Return Me.GetRelatedTableData(relatedType, intermediaryType)
		End Function
	End Class 
End Namespace

File: src/tests/DataBlockASPNETDemo/csharp/Mappings/Book.cs
using System;
using System.Data;
using voidsoft.DataBlock;

namespace BookManager.Mappings
{

    [Serializable()]
    public class Book : TableMetadata
    {
        private DatabaseField[] _fields;

        public Book()
        {
            _fields = new DatabaseField[6];
            _fields[0] = new DatabaseField(DbType.Int32, "BookId", true, true, null);
            _fields[1] = new DatabaseField(DbType.String, "Name", false, false, null);
            _fields[2] = new DatabaseField(DbType.Int32, "Pages", false, false, null);
            _fields[3] = new DatabaseField(DbType.String, "ISBN", false, false, null);
            _fields[4] = new DatabaseField(DbType.String, "Genre", false, false, null);
            _fields[5] = new DatabaseField(DbType.Int16, "Grade", false, false, null);

            this.currentTableName = "Book";

            this.listRelations.Add(new ParentTableRelation("BookAuthors", "BookId", TableRelationCardinality.OneToMany, true));
            this.listRelations.Add(new ManyToManyTableRelation("Author", "BookAuthors", "BookId", "AuthorId"));

        }


        public override DatabaseField[] TableFields
        {
            get
            {
                return _fields;
            }
            set
            {
                _fields = value;
            }
        }

        public System.Int32 BookId
        {
            get
            {
                return (System.Int32)(this.GetField("BookId")).fieldValue;
            }

            set
            {
                this.SetFieldValue("BookId", value);
            }
        }


        public System.String Name
        {
            get
            {
                return (System.String)(this.GetField("Name")).fieldValue;
            }

            set
            {
                this.SetFieldValue("Name", value);
            }
        }


        public System.Int32 Pages
        {
            get
            {
                return (System.Int32)(this.GetField("Pages")).fieldValue;
            }

            set
            {
                this.SetFieldValue("Pages", value);
            }
        }


        public System.String ISBN
        {
            get
            {
                return (System.String)(this.GetField("ISBN")).fieldValue;
            }

            set
            {
                this.SetFieldValue("ISBN", value);
            }
        }


        public System.String Genre
        {
            get
            {
                return (System.String)(this.GetField("Genre")).fieldValue;
            }

            set
            {
                this.SetFieldValue("Genre", value);
            }
        }


        public System.Int16 Grade
        {
            get
            {
                return (System.Int16)(this.GetField("Grade")).fieldValue;
            }

            set
            {
                this.SetFieldValue("Grade", value);
            }
        }

        public Array GetBookAuthors()
        {
            Type relatedType = Type.GetType("BookManager.Mappings.BookAuthors");
            return this.GetRelatedTableData(relatedType);
        }
        public Array GetAuthor()
        {
            Type relatedType = Type.GetType("BookManager.Mappings.Author");
            Type intermediaryType = Type.GetType("BookManager.Mappings.BookAuthors");
            return this.GetRelatedTableData(relatedType, intermediaryType);
        }
    }
}
